aten/src/ATen/Context.h

```diff
@@ -76,6 +77,9 @@ class CAFFE2_API Context {
   bool hasHIP() const {
     return detail::getHIPHooks().hasHIP();
   }
+  bool hasXLA() const {
+    return detail::getXLAHooks() != nullptr;
```

Comment:
Why doesn't this match the other hooks, which are instantiated with versions of the hooks that can return false?

Reply:
HIP/CUDA works different from XLA, as far as "bring up" POV.
PT loads CUDA/HIP when it boots. The notion of has-HIP, has-CUDA is embedded in the build.
We load PT, then install our hooks.
The very same version of PT can load w/out XLA.
We could install a dummy version of the XLA hooks, even though there is no XLA, if that's what you prefer.

Reply:
ya, that's what I prefer.

Also, PT doesn't load CUDA/HIP when it boots, only when code requiring it is exercised.

Reply:
Applied hooks style similar to CUDA/HIP.